src/api/models/publishedDataset.js

```diff
@@ -4,8 +4,18 @@ import omit from 'lodash.omit';
 
 export default (db) => {
     const collection = db.collection('publishedDataset');
+
     collection.insertBatch = documents => chunk(documents, 100).map(data => collection.insertMany(data));
-    collection.findLimitFromSkip = (limit, skip) => collection.find().skip(skip).limit(limit).toArray();
+
+    collection.findLimitFromSkip = (limit, skip) =>
```

Comment:
would be better to add an optional `filter` parameter here and use this method in `findPage` imo